packages/react-button/src/components/Button/useButton.ts

```diff
@@ -14,14 +15,16 @@ export const useButton = (
   options: ComposePreparedOptions,
 ): ButtonState => {
   const buttonRef = useRef<HTMLButtonElement | null>(null);
-
   useImperativeHandle(props.componentRef, () => ({
     focus: () => buttonRef.current?.focus(),
   }));
   useFocusRects(ref as React.RefObject<HTMLElement>);
 
+  const buttonBehaviorProps = useButtonBehavior(props);
+
   return {
     ...props,
+    ...buttonBehaviorProps,
```

Comment:
```suggestion
    ...buttonBehaviorProps,
    ...props,
```

User's input should always win, I suggest to have a dedicated test for this

Reply:
Made the change and added some tests.